mojo-executor-maven-plugin/src/main/java/org/twdata/maven/mojoexecutor/plugin/MojoExecutorMojo.java

```diff
@@ -83,13 +89,47 @@
      */
     private BuildPluginManager pluginManager;
 
-    public void execute() throws MojoExecutionException {
-        if ( mavenProject.getArtifactId().equals( "mojo-executor-test-project-null-maven-project" ))
+    public void execute() throws MojoExecutionException
+    {
+        integrationTestSetup();
+
+        getLog().info( "Executing with maven project " + mavenProject + " for session " + mavenSession );
+        executeMojo( plugin, goal, toXpp3Dom( configuration ),
+                     executionEnvironment( mavenProject, mavenSession, pluginManager ) );
+    }
+
+    /**
+     * Used purely to customise the setups for the integration tests.
+     *
+     * @throws MojoExecutionException
+     */
+    private void integrationTestSetup() throws MojoExecutionException
+    {
+        // Test specific conditionals...
```

Comment:
Maybe a plugin that can execute custom Groovy would make that easier?

Something like ["Executing Groovy Scripts in a Maven Build"](https://books.sonatype.com/mcookbook/reference/ch02s03.html) or [GMaven Plus: Execute scripts](https://github.com/groovy/GMavenPlus/wiki/Examples#execute-scripts).

It's not clear how anyone would actually make use of this functionality if it's restricted to running in a specific artifact ID.